src/App.jsx

```diff
@@ -1,12 +1,133 @@
-import './App.css'
+import "./App.css";
+import { Button } from "./components/Button/Button.jsx";
+import { Card } from "./components/Card/Card.jsx";
+import { useEffect, useState } from "react";
+import axios from "axios";
 
 function App() {
+  const [pokemons, setPokemons] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [pokemonDetails, setPokemonDetails] = useState([]);
+  const [url, setUrl] = useState(
+    "https://pokeapi.co/api/v2/pokemon/?limit=20&offset=0",
+  );
+  const [pagination, setPagination] = useState({
+    next: "",
+    previous: "",
+  });
+
+  useEffect(() => {
+    const controller = new AbortController();
+
+    async function getPokemon() {
+      setError(false);
+      try {
+        setIsLoading(true);
+        const { data } = await axios.get(url, { signal: controller.signal });
+        setPagination({
+          next: data.next,
+          previous: data.previous,
+        });
+        setPokemons(data.results);
+      } catch (error) {
+        // zou hier evt nog wat met de signal kunnen doen van abort controller
+        console.log(error);
+        setError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    getPokemon();
+
+    return () => {
+      controller.abort();
+    };
+  }, [url]);
+
+  useEffect(() => {
+    const controller = new AbortController();
+
+    async function getPokemonDetails() {
+      setError(false);
+      try {
+        setIsLoading(true);
+
+        const results = await Promise.all(
+          pokemons.map(async (pokemon) => {
+            const response = await axios.get(pokemon.url, {
+              signal: controller.signal,
+            });
+            return response.data;
+          }),
+        );
+        setPokemonDetails(results);
+      } catch (error) {
+        // zou hier evt nog wat met de signal kunnen doen van abort controller
+        console.log(error);
+        setError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    getPokemonDetails();
+
+    return () => {
+      controller.abort();
+    };
+  }, [pokemons]);
+
+  function handleClick(buttonLabel) {
```

Comment:
Is `vorige` en `volgende` een _label_, of meer de actie die gedaan moet worden? 🤔